### ISSUE TITLE
Shares always end up with error

### ISSUE THREAD

**Issue by Prochy20** [directus/directus#10701]
### Preflight Checklist

- [X] I have completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).
- [X] I'm on [the latest version of Directus](https://github.com/directus/directus/releases).
- [X] There's [no other issue](https://github.com/directus/directus/issues?q=is%3Aissue) that already describes my problem.

### Describe the Bug

Hi,

whenever I try to share an item via new feature "Shares" it always end up with error - see screenshot:

![image](https://user-images.githubusercontent.com/54104598/147462226-f66f1b74-1750-48fb-a4db-386dc3259e94.png)

My settings:
![image](https://user-images.githubusercontent.com/54104598/147462271-366eb40d-61b3-4a5d-904b-c5dc43e96f40.png)

Generated link: https://my-web-url.domain/admin/shared/ad9df134-91ea-411a-9e3a-9daff0dbe015

Record in directus_shares for this item:
```
INSERT INTO `directus_shares` (`id`, `name`, `collection`, `item`, `role`, `password`, `user_created`, `date_created`, `date_start`, `date_end`, `times_used`, `max_uses`)
VALUES
	('ad9df134-91ea-411a-9e3a-9daff0dbe015', '123456', 'players_wiki', '59', 'c5f9bb47-c1c7-4a7d-a62f-5fc060984826', '$argon2i$v=19$m=4096,t=3,p=1$d/YuybqTg9QXIQWwu4QHcw$Q1JhGFwQpsqNSS45kSpx1dk6l4C7uwKYTtnJHb5RftQ', 'bbb1b8f7-69c7-496b-a15b-47f157d95ef2', '2021-12-27 10:13:27', '2021-12-26 10:13:00', '2021-12-31 10:13:00', 0, 15);
```

What I tried myself:
1) To select a role with "unlimited" permissions
2) To set Max users to 9999
3) To set Date start and Date end year before and after today
4) To give public role unlimited permissions (in therms of reading)
5) Not to set password
6) Not to set dates
7) Not to set Date start & Date end
8) Manually change in DB role_id for administrator role

APP: Directus 9.4.0
OS: Ubuntu 18.04 LTS
MySQL: Ver 14.14 Distrib 5.7.36, for Linux (x86_64) using EditLine wrapper
Node: 14.18.2

### To Reproduce

1) Update Directus from 9.3.0 to 9.4.0
2) Create share on random collection in DB
3) Open generated link in incognito window

### Errors Shown

![image](https://user-images.githubusercontent.com/54104598/147462774-849575d6-59e6-45b2-8c3b-21c8bc779958.png)

### What version of Directus are you using?

9.4.0

### What version of Node.js are you using?

14.18.2

### What database are you using?

MySQL: Ver 14.14 Distrib 5.7.36, for Linux (x86_64) using EditLine wrapper

### What browser are you using?

Google Chrome v 96.0.4664.110

### What operating system are you using?

Desktop: MacOS Monterey 12.1 | Server: Ubuntu 18.04 LTS

### How are you deploying Directus?

Digital Ocean VPS (Using Ubuntu 18.04 LTS)

**Comment by haniefss** [comment]
I can confirm this bug. 

In my case, the Shares panel not even showing any successfully created share. I can confirm there are shares records inside `directus_shares` and when I try to access one of them it shows exactly the same error message as shown above.

**Comment by benhaynes** [comment]
What database/version are you using, @haniefss ?

**Comment by haniefss** [comment]
> What database/version are you using, @haniefss ?

I'm using 9.4.0 (item sharing was shipped with 9.4.0 right) and MariaDB (on aws) for database.

Additional note:

- there's no confirmation when share is created successfully, but I can see them on `directus_shares` table.
- creating share without setting start date & end date will fail.

**Comment by hugopvl** [comment]
I can confirm this bug, its the same for me. database: SQLSERVER

Thanks.
